src/java.base/share/classes/sun/security/pkcs/PKCS8Key.java

```diff
@@ -104,11 +105,28 @@ public PKCS8Key(byte[] input) throws InvalidKeyException {
         }
     }
 
-    private PKCS8Key(byte[] privEncoding, byte[] pubEncoding)
+    /**
+     * Constructor that takes both public and private encodings.  If
+     * publicEncoding is null, a V1 PKCS8 encoding is created; otherwise, V2 is
```

Comment:
This comment: "If publicEncoding is null, a V1 PKCS8 encoding is created"

I don't think this is always true. A V2 encoding could be created if `publicEncoding` is `null` and `decode()` parses the `privateEncoding` and the version is V2.